mathics/builtin/files_io/filesystem.py

```diff
@@ -538,17 +534,16 @@ def apply_with_n(self, name, n, evaluation, options):
             py_n = 1
         else:
             expr = to_expression("DirectoryName", name, n)
-            py_n = n.to_python()
+            py_n = n.to_python(string_quotes=False)
```

Comment:
Isn't `py_n` a Integer?

Reply:
I just looked again at. When it is valid, yes it is an integer. However when it is invalid it could be a string. That said, ths keyword parameter doesn't do anything, so I have removed. it in f44e8ae2589cee